src/Ogooreck.Sample.BusinessLogic.FSharp.Tests/Deciders/BankAccount.fs

```diff
@@ -0,0 +1,69 @@
+﻿module Deciders.BankAccount
+
+open System
+open BankAccountPrimitives
+
+type BankAccountOpened =
+    { BankAccountId: AccountId
+      AccountNumber: AccountNumber
+      ClientId: ClientId
+      CurrencyISOCode: CurrencyCode
+      CreatedAt: DateTimeOffset
+      Version: int64 }
+
+type DepositRecorded =
+    { BankAccountId: AccountId
+      Amount: decimal
+      CashierId: CashierId
+      RecordedAt: DateTimeOffset
+      Version: int64 }
+
+type CashWithdrawnFromATM =
+    { BankAccountId: AccountId
+      Amount: decimal
+      ATMId: AtmId
+      RecordedAt: DateTimeOffset
+      Version: int64 }
+
+type BankAccountClosed =
+    { BankAccountId: AccountId
+      Reason: string
+      ClosedAt: DateTimeOffset
+      Version: int64 }
```

Comment:
Bad news to interminingle versions into state. But then you knew that are on the way to refactring it out eventually :D

Reply:
I know it's not ideal, but I'll fight that battle in other examples 😅

Reply:
I'm confident you'll refactor toward The Right Thing in time :P